packages/block-library/src/paragraph/edit.js

```diff
@@ -113,7 +111,12 @@ function ParagraphBlock( {
 					>
 						<ToggleControl
 							label={ __( 'Drop cap' ) }
-							checked={ !! dropCap }
+							checked={
+								align === ( isRTL() ? 'left' : 'right' ) ||
+								align === 'center'
+									? false
+									: dropCap
```

Comment:
Unchecks the toggle if the option is disabled.
The option is restored to the previous checked or unchecked state if the text alignment is removed.

The amount of times the if else is repeated is becoming a bit ridiculous 😀

Reply:
I left some suggestions above; I hope that helps reduce the number 😄

Reply:
yeah I didn't see your comment until after :)